metricflow/specs/where_filter_dimension.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from typing import List, Sequence
+
+from dbt_semantic_interfaces.call_parameter_sets import (
+    DimensionCallParameterSet,
+    FilterCallParameterSets,
+)
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.references import (
+    DimensionReference,
+    EntityReference,
+)
+from typing_extensions import override
+
+from metricflow.specs.column_assoc import ColumnAssociationResolver
+from metricflow.specs.query_interface import (
+    QueryInterfaceDimension,
+    QueryInterfaceDimensionFactory,
+)
+from metricflow.specs.specs import DimensionSpec
+
+
+class WhereFilterDimension(ProtocolHint[QueryInterfaceDimension]):
```

Comment:
I don't have time to review the new classes right now, so I'm checkpointing here and submitting for the time being.

As a broader note, this PR is really big now. In future I'd recommend splitting it up, something along the lines of:

1. Add protocol classes and one example implementation
2. Cut over other implementations one at a time in separate PRs

Then you'd have like 3 or 4, one of which gets the most action and the others are easy quick reads.

I wouldn't bother doing this here unless there's a big hurry to get one of these things merged without the others, I'm like 80-90% through it. But in future it'll be easier to make progress in chunks.

Reply:
Thanks, Tom! I will do that.